src/Farmer/Builders/Builders.Storage.fs

```diff
@@ -14,13 +15,17 @@ let internal buildKey (ResourceName name) =
             name
     |> ArmExpression.create
 
+type ShareQuotaInGb = int
+
 type StorageAccountConfig =
     { /// The name of the storage account.
       Name : ResourceName
       /// The sku of the storage account.
       Sku : Sku
       /// Containers for the storage account.
-      Containers : (ResourceName * StorageContainerAccess) list}
+      Containers : (ResourceName * StorageContainerAccess) list
+      /// File shares
+      FileShares: (ResourceName * ShareQuotaInGb option) list }
```

Comment:
For simple types like this, I think that a nice way to do that could be to change from a tuple to a full record. It would then be available from the `StorageAccountConfig` as a standard field and could be referenced wherever you want. If you want to avoid the record syntax, a static member on the type might be enough e.g. `create(resourceName:string, ?ShareQuoteInGb:int)`.